File: Java/SingleNumber.java
public class SingleNumber {
    public static int singleNumber(int[] nums) {
        int num = 0;
        for (int i : nums) {
            num ^= i;
        }
        return num;
    }

    public static void main(String args[]){
        int[] nums = {2, 3, 10, 3, 7, 1, 2, 1, 10};
        System.out.println(singleNumber(nums));
    }
}

File: PythonCode/SingleNumber.py
from typing import List


class Solution:
    # Time complexity: O(n)
    # Space complexity: O(1)
    def singleNumber(self, nums: List[int]) -> int:
        val = 0
        for num in nums:
            val ^= num
        return val


if __name__ == '__main__':
    solution = Solution()
    print(solution.singleNumber([2, 2, 1]))         # 1
    print(solution.singleNumber([4, 1, 2, 1, 2]))   # 4
    print(solution.singleNumber([1]))               # 1
